src/launch/automation/terragrunt/functions.py

```diff
@@ -136,48 +141,28 @@ def prepare_for_terragrunt(
     # If the Provider is AZURE there is a prequisite requirement of logging into azure
     # i.e. az login, or service principal is already applied to the environment.
     # If the provider is AWS, we need to assume the role for deployment.
+    provider = provider_config["provider"]
     if provider_config:
-        if provider_config["provider"] == "aws":
+        if provider == "aws":
             assume_role(
                 provider_config=provider_config,
                 repository_name=name,
                 target_environment=target_environment,
             )
-        if provider_config["provider"] == "az":
+        if provider == "az" or provider == "azdo":
```

Comment:
Given that we support [a bunch of providers](https://github.com/nexient-llc/lcaf-component-policy/blob/main/policy/approved_providers.rego#L7-L10) that are prefixed with `az`, this might cover more cases:

```suggestion
        if provider.startswith("az"):
```